master/buildbot/data/forceschedulers.py

```diff
@@ -38,22 +38,29 @@ class ForceSchedulerEndpoint(base.Endpoint):
         /forceschedulers/i:schedulername
     """
 
-    def get(self, resultSpec, kwargs):
+    def findForceScheduler(self, schedulername):
+        # eventually this may be db backed. This is why the API is async
         for sched in self.master.allSchedulers():
-            if sched.name == kwargs['schedulername'] and isinstance(sched, forcesched.ForceScheduler):
-                return defer.succeed(forceScheduler2Data(sched))
-        return defer.succeed(None)
+            if sched.name == schedulername and isinstance(sched, forcesched.ForceScheduler):
+                return defer.succeed(sched)
+
+    @defer.inlineCallbacks
+    def get(self, resultSpec, kwargs):
+        sched = yield self.findForceScheduler(kwargs['schedulername'])
+        if sched is not None:
+            defer.returnValue(forceScheduler2Data(sched))
 
     @defer.inlineCallbacks
     def control(self, action, args, kwargs):
         if action == "force":
-            for sched in self.master.allSchedulers():
-                if sched.name == kwargs['schedulername'] and isinstance(sched, forcesched.ForceScheduler):
-                    try:
-                        res = yield sched.force("user", **args)
-                        defer.returnValue(res)
-                    except forcesched.CollectedValidationError as e:
-                        raise BadJsonRpc2(e.errors, JSONRPC_CODES["invalid_params"])
+            sched = yield self.findForceScheduler(kwargs['schedulername'])
```

Comment:
do we always have `schedulername` here? I can see that the previous code assumed that as well, so my question would be for the old code as well :)